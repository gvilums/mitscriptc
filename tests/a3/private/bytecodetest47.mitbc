function
{
	functions =
	[
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		}
	],
	constants = [3, "This should be printed.", false, "This should not be printed.", 0, 1],
	parameter_count = 0,
	local_vars = [],
	local_ref_vars = [],
	free_vars = [],
	names = [x, print, input, intcast],
	instructions = 
	[
		load_func	0
		alloc_closure	0
		store_global	1
		load_func	1
		alloc_closure	0
		store_global	2
		load_func	2
		alloc_closure	0
		store_global	3
		load_const	0
		store_global	0
		load_global	0
		load_const	0
		eq
		not
		if	5
		load_global	1
		load_const	1
		call	1
		pop
		load_const	2
		not
		if	6
		load_global	1
		load_const	3
		call	1
		pop
		goto	5
		load_global	1
		load_const	1
		call	1
		pop
		load_global	0
		load_const	4
		gt
		not
		if	10
		load_global	1
		load_global	0
		call	1
		pop
		load_global	0
		load_const	5
		sub
		store_global	0
		goto	-13
	]
}