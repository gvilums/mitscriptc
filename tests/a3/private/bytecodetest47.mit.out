This should be printed.
This should be printed.
3
2
1
